File: D5/sumOfDivisors.cpp
// Created: 13/Aug/23 at 11 : 43
#include<bits/stdc++.h>
using namespace std;
#define ll long long

const int M = 1e9 +7;
ll c2(ll n){
    return n%M*((n - 1)%M)%M*((M + 1)/2)%M;
}

int main(){
ios_base::sync_with_stdio(0);cin.tie(nullptr);

    ll n, ans = 0;
    cin >> n;
    for(ll i = 1; i <= n ;++i){
        ll r = n/(n/i);
        ans += n/i%M*(c2(r + 1) - c2(i)+M)%M;
        i = r;
    }
    cout << ans%M;


return 0;
}

File: D5/sumOfDivisors.py
n = int(input())
mod = 10**9 + 7
ans = 0
i = 1
while i <= n:
    r = n // i
    ans = (ans + (n // i) * (r * (r + 1) // 2 - i * (i - 1) // 2)) % mod
    i = r + 1
print(ans)
